workflows/pipe-common/scripts/autoscale_sge.py

```diff
@@ -951,26 +1217,45 @@ def _retrieve_preference(pipe, preference, default_value):
         if 'CP_CAP_AUTOSCALE_WORKERS' in os.environ else 3
     log_verbose = os.environ['CP_CAP_AUTOSCALE_VERBOSE'].strip().lower() == "true" \
         if 'CP_CAP_AUTOSCALE_VERBOSE' in os.environ else False
+    hybrid_autoscale = os.environ['CP_CAP_AUTOSCALE_HYBRID'].strip().lower() == "true" \
+        if 'CP_CAP_AUTOSCALE_HYBRID' in os.environ else False
+    autoscale_max_core_per_instance = int(os.getenv('CP_CAP_AUTOSCALE_HYBRID_MAX_CORE_PER_NODE', sys.maxint))
+    instance_family = os.getenv('CP_CAP_AUTOSCALE_HYBRID_FAMILY',
+                                CloudPipelineInstanceHelper.get_family_from_type(cloud_provider, instance_type))
+
+    pipeline_version = get_pipeline_version()
+
+    pipe = PipelineAPI(api_url=pipeline_api, log_dir='/common/workdir/.pipe.log')
+
+    instance_helper = CloudPipelineInstanceHelper(cloud_provider=cloud_provider, region_id=region_id,
+                                                  instance_family=instance_family, master_instance_type=instance_type,
+                                                  pipe=pipe, pipeline_version=pipeline_version,
+                                                  max_core_per_instance=autoscale_max_core_per_instance)
+
+    max_instance_cores = instance_helper.get_max_allowed(price_type, hybrid_autoscale).cpu
+    max_cluster_cores = max_instance_cores * max_additional_hosts + instance_cores
```

Comment:
We should count permanent worker cores here as well as additional ones. Could you please look through the pull request once again maybe there are similar cases where we doesn't handle permanent worker properly? I've looked through and haven't found anything. But maybe you will.

Reply:
I also checked and didn't find anything else